src/v/kafka/server/handlers/alter_partition_reassignments.cc

```diff
@@ -252,7 +251,10 @@ ss::future<response_ptr> alter_partition_reassignments_handler::handle(
             }
         }
 
-        resp.data.responses.push_back(topic_response);
+        // Insert into the response if there were valid partitions
+        if (std::distance(topic.partitions.begin(), valid_partitions_end) > 0) {
```

Comment:
Is this the same as `!topic_partitions.empty()`?

Reply:
I see your point; however, the `validate_partitions` method does lots of iterator magic so I would rather be safe by specifically checking the distance between iterators.